tests/config_snmp_test.py

```diff
@@ -0,0 +1,814 @@
+        assert db.cfgdb.get_entry("SNMP", "CONTACT") == {}
+
+    def test_config_snmp_contact_add_del_new_contact(self):
+        db = Db()
+        runner = CliRunner()
+        with mock.patch('utilities_common.cli.run_command') as mock_run_command:
+            result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["add"],
+                                    ["testuser", "testuser@contoso.com"], obj=db)
+            print(result.exit_code)
+            print(result.output)
+            assert result.exit_code == 0
+            assert result.output == config_snmp_contact_add_del_new_contact
+            assert db.cfgdb.get_entry("SNMP", "CONTACT") == {"testuser": "testuser@contoso.com"}
+
+            result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["del"],
+                                    ["testuser"], obj=db)
+            print(result.exit_code)
+            print(result.output)
+            assert result.exit_code == 0
+            assert 'SNMP contact testuser removed from configuration' in result.output
+            assert db.cfgdb.get_entry("SNMP", "CONTACT") == {} 
+
+    # Add snmp contact tests
+    def test_config_snmp_contact_add_with_existing_contact(self):
+        db = Db()
+        runner = CliRunner()
+        with mock.patch('utilities_common.cli.run_command') as mock_run_command:
+            result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["add"],
+                                    ["testuser", "testuser@contoso.com"], obj=db)
+            print(result.exit_code)
+            print(result.output)
+            assert result.exit_code == 0
+            assert result.output == config_snmp_contact_add_del_new_contact
+            assert db.cfgdb.get_entry("SNMP", "CONTACT") == {"testuser": "testuser@contoso.com"}
+
+            result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["add"],
+                                                     ["blah", "blah@contoso.com"], obj=db)
+            print(result.exit_code)
+            assert result.exit_code == 1
+            assert 'Contact already exists.  Use sudo config snmp contact modify instead' in result.output
+
+    # Delete snmp contact tests
+    def test_config_snmp_contact_del_without_existing_contact(self):
+        db = Db()
+        runner = CliRunner()
+        with mock.patch('utilities_common.cli.run_command') as mock_run_command:
+            result = runner.invoke(config.config.commands["snmp"].commands["contact"].commands["add"],
+                                    ["testuser", "testuser@contoso.com"], obj=db)
+            print(result.exit_code)
+            print(result.output)
+            assert result.exit_code == 0
+            assert result.output == config_snmp_contact_add_del_new_contact
+            assert db.cfgdb.get_entry("SNMP", "CONTACT") == {"testuser": "testuser@contoso.com"}
```

Comment:
same comment as above, is this add user required here?

Reply:
I've changed the function name to be more clear to this: 
test_config_snmp_contact_del_new_contact_when_contact_exists
I just want to test adding a deleting a contact.  
But I want there to be a contact already configured so I verify if there is a contact configure the del will fail because the wrong contact was trying to be deleted.